backend/src/connector/mod.rs

```diff
@@ -1 +1,267 @@
+            })
+            .filter_map(|(name, hostname)| {
+                if hostname.is_err() {
+                    tracing::warn!("couldn't change port for {name}. Client won't be created");
+                }
+                hostname
+                    .ok()
+                    .map(|hostname| (name, Client::try_new_http(&hostname.to_string())))
+            })
+            .filter_map(|(name, client)| {
+                if client.is_err() {
+                    tracing::warn!("couldn't create client for {hostname}");
+                }
+                client.ok().map(|client| {
+                    (
+                        name.clone(),
+                        Arc::new(client.with_context(context.clone())) as Arc<ApiInterface>,
+                    )
+                })
+            })
+            .collect();
+        let client = Box::new(
+            Client::try_new_http(&hostname.to_string()).change_context(ClientError::InitClient)?,
+        );
+        Ok(Self {
+            hostname: bob_data.hostname,
+            main: Arc::new(client.with_context(context)),
+            cluster,
+        })
+    }
+
+    /// Probes connection to the Bob cluster
+    ///
+    /// Returns `StatusCode::OK` on success
+    ///
+    /// # Errors
+    ///
+    /// The function fails if there was an error during creation of request
+    /// It shouldn't happen on normal behaviour
+    ///
+    pub async fn probe(&self) -> Result<StatusCode, ClientError> {
+        match self
+            .main
+            .get_nodes()
+            .await
+            .change_context(ClientError::Inaccessible)?
+        {
+            api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(_) => Ok(StatusCode::OK),
+            api::GetNodesResponse::PermissionDenied => Err(ClientError::PermissionDenied.into()),
+        }
+    }
+
+    /// Probes Node's connection
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if no client present for the specified Node's name,
+    /// the client was unable to receive response or the client doesn't have authority to do request.
+    pub async fn probe_socket(&self, name: &NodeName) -> Result<StatusCode, ClientError> {
+        if let Some(client) = self.cluster.get(name) {
+            match client
+                .get_nodes()
+                .await
+                .change_context(ClientError::Inaccessible)?
+            {
+                api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(_) => {
+                    Ok(StatusCode::OK)
+                }
+                api::GetNodesResponse::PermissionDenied => {
+                    Err(ClientError::PermissionDenied.into())
+                }
+            }
+        } else {
+            Err(ClientError::NoClient.into())
+        }
+    }
+
+    #[must_use]
+    pub fn context(&self) -> &ClientContext {
+        self.main.context()
+    }
+
+    #[must_use]
+    pub fn api(&self) -> &Arc<ApiInterface> {
```

Comment:
I think, here the reference to `ApiInterface` can be returned. No need in additional indirection through `Arc`

Reply:
Also additional function to access the api of other node in cluster can be useful:

```Rust
pub fn api_main(&self) -> &ApiInterface {...}
pub fn api_secondary(&self, node_name: &str) -> &ApiInterface {...}
```